core/codegen/src/derive/from_param.rs

```diff
@@ -0,0 +1,43 @@
+use devise::Support;
+use proc_macro2::TokenStream;
+use quote::quote;
+use devise::*;
+use devise::ext::SpanDiagnosticExt;
+use crate::exports::*;
+
+pub fn derive_from_param(input: proc_macro::TokenStream) -> TokenStream {
+    DeriveGenerator::build_for(input, quote!(impl<'a> #_request::FromParam<'a>))
+        .support(Support::Enum)
+        .validator(ValidatorBuild::new()
+                   .fields_validate(|_, fields| {
+                       if !fields.is_empty() {
+                           return Err(fields.span().error("Only empty enums are accepted"));
+                       }
+
+                       Ok(())
+                   })
+        )
+        .inner_mapper(MapperBuild::new()
+                      .enum_map(|_, data| {
+                          let mut matches = vec![];
+
+                          for field in data.variants() {
+                              let field_name = &field;
+                              matches.push(quote!(
+                                  stringify!(#field_name) => Ok(Self::#field_name),
+                              ))
+                          }
+
+                          quote! {
+                              type Error = &'a str;
+                              fn from_param(param: &'a str) -> Result<Self, Self::Error> {
+                                  match param {
+                                      #(#matches)*
+                                      _ => Err("Failed to find enum")
+                                  }
+                              }
+                          }
+                      })
```

Comment:
Can you take a look at the indentation here?

Reply:
I don't see any issue with it. I checked with the auto-indent in my IDE and with `cargo fmt --check`.

Reply:
Rocket intentionally disables rustfmt (`cargo fmt`), since it doesn't handle some of the more complex trait bounds well. I suspect your IDE's auto-indent is based on rustfmt as well (I don't think there is another well built option out there yet), so that's likely why it's not picking up on the issues here.

The specific issue here is that not all the indentations are the same size (This is extremely obvious in my editor, since I have indent guides turned on). If you would like, I can push a fixed version to this PR.

Reply:
Even after enabling the indent guide, I don't see it. So it would be nice if you can do it.